src/main/java/com/washingtonpost/kafka/Configuration.java

```diff
@@ -51,26 +69,61 @@ private Configuration() {
                 if (body == null || body.isEmpty()) {
                     throw new Exception(configUrl+" response is empty");
                 }
-                logger.info(body);
+                LOGGER.info(body);
                 config = mapper.readValue(body, Config.class);
             }
 
-            if (config.kafka.bootstrapServers == null && config.kafka.host != null && config.kafka.port != null) {
-                config.kafka.bootstrapServers = fetchKafkaIPs(config.kafka.host, config.kafka.port);
-            }
+            this.useIpAddressForBootstrapServers();
+            
         } catch (Exception e) {
-            logger.error(e);
+            LOGGER.error(e);
+        }
+    }
+
+    void useIpAddressForBootstrapServers() {
+        if (config.kafka.bootstrapServers == null && config.kafka.host != null && config.kafka.port != 0) {
+            if (this.useIp(config.kafka.useIp)) {
+                config.kafka.bootstrapServers = FETCH_KAFKA_IPS(
+                    this.findHosts(config.kafka.host).collect(Collectors.toList()), 
+                    config.kafka.port);
+            } else {
+                config.kafka.bootstrapServers = this.appendPort(config.kafka.host, config.kafka.port);
+            }
         }
     }
 
+    boolean useIp(final String value) {
+        return value == null || BooleanUtils.isTrue(BooleanUtils.toBooleanObject(value));
```

Comment:
leading `value == null` should change to add blank string as the java object for Kafka will initialize to empty string